### ISSUE TITLE
[Bug]: When the URL creation fails, the user is not told why

### ISSUE THREAD

**Issue by XLTechie** [nvdaes/urlShortener#10]
### Summary

If for some reason a 406 error is generated when creating a new shortened URL, the user is not told that an error has occurred, or what it is, and an exception is not caught.

While creating a new shortened URL, I got an NVDA error sound (because I'm running NVDA alpha), and was taken back to the list of existing URLs.

Because of that error sound, I knew there was a problem, and checked the log.

### To reproduce

Attempt to create a new shortened URL.
In the Custom URL field, enter something with letters, numbers, a dash, and a period. I'm not sure which character caused the problem for me, but I had something like "wordWord-2023.1".

### Expected behavior

In what I think the order of importance is for fixing:

1. An error dialog should have appeared to tell me there was a problem, before the new URL window was closed.
2. Instead of going back to the list of URLs, and losing my input, it should return to the new URL dialog with my already entered information, and allow changing of parameters to get a successful creation. In this case, it is likely that it didn't like some character (a period or dash) in the `Custom URL` field.
3. There is an unhandled exception caused by the 406 error not returning the expected type. This exception should probably be prevented, and instead a user error be caused, or maybe a debug message logged.
4. The error might have contained information about the problem if there was any. If a 406 error was returned from the API server, if it had some kind of error text in its body, that should have been captured and reported to the user.

### Relevant log output

```shell
DEBUGWARNING - external:globalPlugins.urlShortener.isGd.IsGd.shortenUrl (02:12:37.432) - MainThread (2992):
HTTP Error 406: Not Acceptable
ERROR - unhandled exception (02:12:37.432) - MainThread (2992):
Traceback (most recent call last):
  File "C:\Users\luke\AppData\Roaming\nvda\addons\urlShortener\globalPlugins\urlShortener\urlsGui.py", line 237, in onNew
    if not urlMetadata.shortenedUrl:
AttributeError: 'NoneType' object has no attribute 'shortenedUrl'
```

### Version

2.0.0

### System configuration

NVDA: alpha, 2023.2.0.27996
Windows 10, 22H2

**Comment by nvdaes** [comment]
Hi Luke, I'll look at this in the next days. Of course, if you want to create a PR, and even if you wish to be mentioned as an author, feel free to do so, since you have provided valuable feedback. Otherwise I'll see what can be done.

**Comment by XLTechie** [comment]
Hi Noelia

I won't have time to study the internals of the add-on for the next few days, 
but if I do before you work on this, I will submit a PR.

Unrelated, I was going to submit a PR for your issue form, because of some 
English spelling/grammar issues, probably on the "placeholder" fields. But when 
I tried, I found that you don't have a .github/ISSUE_TEMPLATE directory, so I 
guess you are doing the custom form you have through the interface instead of 
through YAML.

**Comment by nvdaes** [comment]
Hi Luke: I may fix the bug in the add-on today, or you may do it before than me. All is right. About issue form, I have a .github repo. Feel free to clone it and fix grammar mistakes or whatever; thanks.Enviado desde mi iPhoneEl 23 abr 2023, a las 9:00, Luke Davis ***@***.***> escribió:﻿
Hi Noelia

I won't have time to study the internals of the add-on for the next few days, 
but if I do before you work on this, I will submit a PR.

Unrelated, I was going to submit a PR for your issue form, because of some 
English spelling/grammar issues, probably on the "placeholder" fields. But when 
I tried, I found that you don't have a .github/ISSUE_TEMPLATE directory, so I 
guess you are doing the custom form you have through the interface instead of 
through YAML.

**Comment by nvdaes** [comment]
My pan is to remove de new URL dialog, and put the edit fields and a button to create the new URL without opening an extra dialog. In this way, the focus can be placed on the URLs list or in the first edit box, deppending of the result after pressing the button. Also, if an error dialog is closed on the New Url modal dialog, the New URL dialog maybe closed and focus is placed in the main dialog. So I think that this is the easier solution. I may do this even the next weekend or before. CheersEnviado desde mi iPhoneEl 23 abr 2023, a las 9:43, Noelia Ruiz Martínez ***@***.***> escribió:﻿
Hi Luke: I may fix the bug in the add-on today, or you may do it before than me. All is right. About issue form, I have a .github repo. Feel free to clone it and fix grammar mistakes or whatever; thanks.Enviado desde mi iPhoneEl 23 abr 2023, a las 9:00, Luke Davis ***@***.***> escribió:﻿
Hi Noelia

I won't have time to study the internals of the add-on for the next few days, 
but if I do before you work on this, I will submit a PR.

Unrelated, I was going to submit a PR for your issue form, because of some 
English spelling/grammar issues, probably on the "placeholder" fields. But when 
I tried, I found that you don't have a .github/ISSUE_TEMPLATE directory, so I 
guess you are doing the custom form you have through the interface instead of 
through YAML.

**Comment by nvdaes** [comment]
Fixed in #11
